sesman/session.c

```diff
@@ -513,33 +508,30 @@ session_start_fork(tbus data, tui8 type, struct SCP_SESSION *s)
          *  $OpenBSD: session.c,v 1.252 2010/03/07 11:57:13 dtucker Exp $
          *  with some ideas about BSD process grouping to xrdp
          */
-        pid_t bsdsespid = g_fork();
 
-        if (bsdsespid == -1)
+        /**
+         * Create a new session and process group since the 4.4BSD
+         * setlogin() affects the entire process group
+         */
+        if (g_setsid() < 0)
         {
+            log_message(LOG_LEVEL_ERROR,
+                        "setsid failed - pid %d", g_getpid());
         }
-        else if (bsdsespid == 0) /* BSD session leader */
-        {
-            /**
-             * Create a new session and process group since the 4.4BSD
-             * setlogin() affects the entire process group
-             */
-            if (g_setsid() < 0)
-            {
-                log_message(LOG_LEVEL_ERROR,
-                            "setsid failed - pid %d", g_getpid());
-            }
 
-            if (g_setlogin(s->username) < 0)
-            {
-                log_message(LOG_LEVEL_ERROR,
-                            "setlogin failed for user %s - pid %d", s->username,
-                            g_getpid());
-            }
+        if (g_setlogin(s->username) < 0)
+        {
+            log_message(LOG_LEVEL_ERROR,
+                        "setlogin failed for user %s - pid %d", s->username,
+                        g_getpid());
         }
-
-        g_waitpid(bsdsespid);
 #endif
+        g_delete_wait_obj(g_term_event);
+        g_tcp_close(g_sck);
```

Comment:
Why is this code moved after the FreeBSD specific code? It's better not to hold resources longer than they are needed.